### ISSUE TITLE
Does location always have to be long and not float?

### ISSUE THREAD

**Issue by aniqueakhtar** [facebookresearch/SparseConvNet#154]
`def pad_with_batch_idx(x,idx): #add a batch index to the list of coordinates
    return torch.cat([x,torch.LongTensor(x.size(0),1).fill_(idx)],1)

def batch_location_tensors(location_tensors):
    a=[]
    for batch_idx, lt in enumerate(location_tensors):
        if lt.numel():
            a.append(pad_with_batch_idx(lt,batch_idx))
    return torch.cat(a,0)`

I am referring to this piece of code. in the padding, I have my `x` as a float. I do not want to convert it into a long because my location (x) are normalized between 0 and 1. 

Please suggest what would be the way to go about it? Should I convert `idx` into a float too and then append them or should I change my `x` to a long?

Thank you

**Comment by btgraham** [comment]
Locations must be between 0 and the spatial_size -1. Float values should be multipled by a scale factor (i.e. x20 to get 5cm-cubed voxels) in the data loader, and rounded to the nearest integer.
